extensions/fairifier-rdf-extension/dtls/fairifier/RdfSkeletonJsonTransformerImpl.java

```diff
@@ -1,26 +1,48 @@
+
 package org.dtls.fairifier;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
-public class RdfSkeletonJsonTransformerImpl implements RdfSkeletonTransformer{
-    private Model model = new Model();
-    
-    @Override
-    public void transform(String model) {
-        this.model.setJson(model);
-    }
+import com.fasterxml.jackson.databind.ObjectMapper;
 
-    @Override
-    public String getModelAsJsonString() {
-        return this.model.getJson();
-    }
+public class RdfSkeletonJsonTransformerImpl implements RdfSkeletonTransformer {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private Model model;
 
     @Override
-    public void setModelMetadata(SkeletonMetadata metadata) {
+    public Model read(Path path)
+            throws IOException {
+        String projectId = path.getFileName().toString().split("\\.")[0];
+
+        this.model = new Model();
+        String json = new String(Files.readAllBytes(path));
+        this.model.setJson(json);
+        Path metadataPath = Paths
+                .get(path.getParent().toString() + File.separator + projectId + ".metadata.skeleton.json");
```

Comment:
I still think we can refactor this. See [my comment on PR#36](https://github.com/DTL-FAIRData/FAIRifier/pull/36#discussion_r141267536) for details.

Reply:
You forgot to include this one in your refactor steps.